chainio/clients/builder.go

```diff
@@ -104,7 +104,7 @@ func BuildAll(
 
 }
 
-func (config *BuildAllConfig) buildElClients(
+func (config *BuildAllConfig) BuildElClients(
```

Comment:
Why do you need this method exposed? Personally find it a bit weird given that the BuildAllConfig takes in AVS contract addresses. I'd prefer we expose a BuildElClients that takes in a core eigenlayer contract address, such as the delegation manager address.

Also if you're already putting together the BuildAllConfig, might as well just build everything, even if you just need the EL clients.

Reply:
Its connected to `InstrumentedClient`. The reason to expose this is because `BuildAll` build clients using `eth.NewClient` which in our case just creates duplicate clients. I would like to be able to pass wrapped eth.Client(`InstrumentedClient`) to achieve results of `BuildAll` but with my own clients

Reply:
> Its connected to `InstrumentedClient`. The reason to expose this is because `BuildAll` build clients using `eth.NewClient` which in our case just creates duplicate clients. I would like to be able to pass wrapped eth.Client(`InstrumentedClient`) to achieve results of `BuildAll` but with my own client instances

Reply:
I see, makes sense to expose these I guess for people to use them, although I'd still prefer we clear the TODO on top of Clients struct, and make a different BuildALL for people who want instrumented client and another for people who don't want metrics. Perhaps easiest is only to expose the building blocks like this, but my main concern was just that the config struct is then a bit confusing. I'm fine with merging these changes for now if it helps, but if you end up creating other useful high-level constructors, please create new PRs with those. :)

Reply:
Sure) Will resolve conflicts in a few days.

Reply:
Done